hazelcast-client/src/test/java/com/hazelcast/client/proxy/ProxyManagerTest.java

```diff
@@ -123,4 +126,19 @@ public void testNextAddressToSendCreateRequestOnMultipleLiteMembers() {
 
         return instances;
     }
+
+    @Test(expected = OperationTimeoutException.class)
+    public void testProxyCreateTimeout_whenClusterIsNotReachable() {
+        HazelcastInstance instance = factory.newHazelcastInstance();
+        ClientConfig config = new ClientConfig();
+        config.setProperty(ClientProperty.INVOCATION_TIMEOUT_SECONDS.getName(), "1");
+        HazelcastInstance client = factory.newHazelcastClient(config);
+        instance.shutdown();
+        try {
+            client.getMap("test");
+        } catch (RuntimeException e) {
+            client.shutdown();
```

Comment:
It may be better not to do this shutdown here but do it in After test method just in case if the OperationTimeoutException exception is emitted from client.shutdown(), then the test will pass even though that was not the intent.